Memmy-App/memmy
# 647 show hide read button

### PR Creator Checklist

Ensure you've checked the following before submitting your PR:


### Summary

> Please provide a summary of what your PR does

This addresses #647 . Details of implementation deviating from the ticket description were discussed with @ktgd .

Adds a Show Hide Read Button setting to enable/disable the visibility of the Hide Read FAB. Included a footer detailing what the Hide Read Posts on Feed and Show Hide Read Button settings do.

### Screenshots

> If the UI has been changed, include screenshots. 
> We will not review your PR without screenshots if they are applicable.

![Simulator Screenshot - iPhone 14 Pro Max - 2023-07-17 at 09 06 04](https://github.com/Memmy-App/memmy/assets/3818037/ec872db2-631a-4b05-addb-c99c499fc1d9)

### Test Plan

> Please documemnt the steps required to test your PR

1. View the feed as normal. Any read posts should remain visible after refreshing the feed.
2. Enable the Hide Read Posts on Feed setting. Repeat step 1. Now any read posts should disappear on refreshing of the feed, but the Eye FAB should not appear.
3. Enable the Show Hide Read Button. Returning to the feed should now show the Eye FAB button. After reading posts, pressing the FAB should remove read posts from the feed without refresh.
